cluster-autoscaler/integration/framework.go

```diff
@@ -315,39 +338,84 @@ func getDeploymentObject(replicas int32, resourceCpu string, resourceMemory stri
 							},
 							Resources: v1.ResourceRequirements{
 								Requests: v1.ResourceList{
-									v1.ResourceCPU:    resource.MustParse(resourceCpu),
-									v1.ResourceMemory: resource.MustParse(resourceMemory),
+									v1.ResourceCPU:    resourceCPU,
+									v1.ResourceMemory: resourceMemory,
 								},
 							},
 						},
 					},
+					Tolerations: tolerations,
 				},
 			},
 		},
 	}
 	return deployment
 }
 
-func (driver *Driver) deployWorkload(replicas int32, workloadName string) error {
-	deployment := getDeploymentObject(replicas, fmt.Sprint(cpuResource-1), mediumMemory, workloadName)
+func (driver *Driver) deployWorkload(replicas int32, workloadName string, canTolerateTaintPlacedOnInitialNodes bool) error {
+	// TODO(himanshu-kun): Remove such a dependency on approximating system components space . This changes over time.
+	assumedSystemComponentsUsedSpace := *resource.NewMilliQuantity(1000, resource.DecimalSI)
+	cpuRequested := cpuResource.DeepCopy()
+	cpuRequested.Sub(assumedSystemComponentsUsedSpace)
+	tolerationsToInitialNodeTaint := []v1.Toleration{
```

Comment:
Can we have this as a separate variable? The same variable is used in other methods as well

Reply:
addressed in [a8a7cef](https://github.com/gardener/autoscaler/pull/187/commits/a8a7cef0877b589973075d0e53b70f2974810ab9)